plugin-build/src/main/kotlin/io/sentry/android/gradle/instrumentation/remap/RemappingInstrumentable.kt

```diff
@@ -0,0 +1,38 @@
+@file:Suppress("UnstableApiUsage")
+
+package io.sentry.android.gradle.instrumentation.remap
+
+import com.android.build.api.instrumentation.ClassContext
+import io.sentry.android.gradle.instrumentation.ClassInstrumentable
+import io.sentry.android.gradle.instrumentation.SpanAddingClassVisitorFactory
+import org.objectweb.asm.ClassVisitor
+import org.objectweb.asm.commons.ClassRemapper
+import org.objectweb.asm.commons.SimpleRemapper
+
+class RemappingInstrumentable : ClassInstrumentable {
+
+    companion object {
+        private val mapping = mapOf(
+            "java/io/FileInputStream" to "io/sentry/instrumentation/file/SentryFileInputStream",
+            "java/io/FileOutputStream" to "io/sentry/instrumentation/file/SentryFileOutputStream"
+        )
+    }
+
+    override val fqName: String get() = "remapper"
+
+    override fun getVisitor(
+        instrumentableContext: ClassContext,
+        apiVersion: Int,
+        originalVisitor: ClassVisitor,
+        parameters: SpanAddingClassVisitorFactory.SpanAddingParameters
+    ): ClassVisitor =
+        ClassRemapper(originalVisitor, SimpleRemapper(mapping))
+
+    override fun isInstrumentable(data: ClassContext): Boolean {
+        return when {
+            data.currentClassData.className.startsWith("io.sentry")
+                && !data.currentClassData.className.startsWith("io.sentry.android.roomsample") -> false
```

Comment:
this is just for the sample to work, probably shouldn't be in the production code.. maybe worth introducing ignore list config in the gradle plugin, where users can specify packagenames to ignore?

Reply:
yeah, but I'd rather take it as feature request, not really part of this PR.
we could rename the sample package if necessary to avoid this.
on Java, we special case these packages btw https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/SentryStackTraceFactory.java#L41-L46
because of our samples and mobile app.